### PR TITLE
add python api example and update doc

### PR DESCRIPTION
**Pull request by J-shang**
A jupyter notebook example about `Experiment.start()` and `Experiment.connect()`.

### CODE REVIEW

**Review by liuzhe-lz — APPROVED** [review]

File: dependencies/required.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ scikit-learn>=0.23.2
 websockets
 filelock
 prettytable
+ipython
```
Comment (QuanluZhang) [inline dependencies/required.txt:16]: what is this file used for?

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst [hunk 2]
```diff
@@ -90,6 +90,213 @@ Below is an example for this new launching approach. You can also find this code
...
+.. parsed-literal::
+
+    [2021-02-22 12:29:01] Tuner not set, wait for connect...
+
+2. Connect Experiment
+"""""""""""""""""""""
+
+.. code:: ipython3
+
+    experiment.connect(8080)
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/HowToLaunchFromPython.rst:219]: which one is correct? `Experiment.connect()`?

File: nni/experiment/experiment.py [hunk 3]
```diff
@@ -316,6 +323,33 @@ def get_experiment_profile(self):
         resp = self._experiment_rest_get(self.port, '/experiment')
         return resp
 
+    def get_experiment_metadata(self, exp_id: str):
+        """
+        Return experiment metadata with specified exp_id as a dict.
+
+        Returns
+        ----------
```
Comment (QuanluZhang) [inline nni/experiment/experiment.py:331]: Returns
--------

File: nni/experiment/experiment.py [hunk 4]
```diff
@@ -329,13 +363,13 @@ def export_data(self):
         return [TrialResult(**trial_result) for trial_result in resp]
 
     def _get_query_type(self, key: str):
-        if key == 'trial_concurrency':
+        if key == 'trialConcurrency':
```
Comment (QuanluZhang) [inline nni/experiment/experiment.py:366]: so the previous version is a bug?

Comment (J-shang) [reply nni/experiment/experiment.py:366]: yes

File: docs/en_US/Tutorial/HowToLaunchFromPython.rst (hunk 2 above)
Comment (J-shang) [reply docs/en_US/Tutorial/HowToLaunchFromPython.rst:219]: for now, `experiment.connect()`.

File: nni/experiment/experiment.py (hunk 4 above)
Comment (QuanluZhang) [reply nni/experiment/experiment.py:366]: haha

File: nni/experiment/experiment.py [hunk 5]
```diff
@@ -204,7 +204,7 @@ def run(self, port: int = 8080, debug: bool = False) -> bool:
         finally:
             self.stop()
 
-    def connect_experiment(self, port: int):
+    def connect(self, port: int):
```
Comment (QuanluZhang) [inline nni/experiment/experiment.py:207]: suggest to decorate it with `@classmethod` or `@staticmethod`

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**